src/overlays/actors/ovl_En_Jg/z_en_jg.c

```diff
@@ -74,70 +133,896 @@ static DamageTable D_80B75858 = {
+        if ((messageState == 5) && (func_80147624(globalCtx))) {
+            globalCtx->msgCtx.msgMode = 0x43;
+            globalCtx->msgCtx.unk12023 = 4;
+            this->flags &= ~FLAG_LOOKING_AT_PLAYER;
+            this->animationIndex = EN_JG_ANIMATION_WALK;
+            func_8013BC6C(&this->skelAnime, sAnimations, this->animationIndex);
+            this->actionFunc = EnJg_Walk;
+        }
+    } else if (this->animationIndex == EN_JG_ANIMATION_WALK) {
+        Math_ApproachF(&this->actor.speedXZ, 0.0f, 0.2f, 1.0f);
+        EnJg_CheckIfTalkingToPlayerAndHandleFreezeTimer(this, globalCtx);
+    }
+}
+
+void EnJg_Walk(EnJg* this, GlobalContext* globalCtx) {
+    s16 yRotation;
+    f32 distSQ;
+
+    if (this->path != NULL) {
+        yRotation = EnJg_GetWalkingYRotation(this->path, this->currentPoint, &this->actor.world.pos, &distSQ);
+        if (this->actor.bgCheckFlags & 8) {
+            yRotation = this->actor.wallYaw;
+        }
+
+        Math_SmoothStepToS(&this->actor.world.rot.y, yRotation, 4, 0x3E8, 1);
+        this->actor.shape.rot.y = this->actor.world.rot.y;
+
+        if (EnJg_ReachedPoint(this, this->path, this->currentPoint)) {
+            if (this->currentPoint >= (this->path->count - 1)) {
+                // Force the elder to walk in place
+                this->animationIndex = EN_JG_ANIMATION_WALK;
+                this->actionFunc = EnJg_AlternateTalkOrWalkInPlace;
+            } else {
+                this->currentPoint++;
+                Math_ApproachF(&this->actor.speedXZ, 0.5f, 0.2f, 1.0f);
+            }
+        } else {
+            Math_ApproachF(&this->actor.speedXZ, 0.5f, 0.2f, 1.0f);
+        }
+    }
+
+    EnJg_CheckIfTalkingToPlayerAndHandleFreezeTimer(this, globalCtx);
+}
+
+void EnJg_Talk(EnJg* this, GlobalContext* globalCtx) {
+    u8 messageState = Message_GetState(&globalCtx->msgCtx);
+    s16 currentFrame = this->skelAnime.curFrame;
+    s16 lastFrame = Animation_GetLastFrame(sAnimations[this->animationIndex].animationSeg);
+    u16 temp;
+
+    if ((this->animationIndex == EN_JG_ANIMATION_SURPRISE_START) && (currentFrame == lastFrame)) {
+        this->animationIndex = EN_JG_ANIMATION_SURPRISE_LOOP;
+        func_8013BC6C(&this->skelAnime, sAnimations, this->animationIndex);
+    }
+
+    if ((messageState == 5) && (func_80147624(globalCtx))) {
+        temp = this->textId;
+        if ((temp == 0xDB4) || (temp == 0xDB5) || (temp == 0xDC4) || (temp == 0xDC6)) {
+            // There is nothing more to say after these lines, so end the current conversation.
+            globalCtx->msgCtx.msgMode = 0x43;
+            globalCtx->msgCtx.unk12023 = 4;
+            this->flags &= ~FLAG_LOOKING_AT_PLAYER;
+            this->actionFunc = EnJg_SetupWalk;
+            return;
+        }
+
+        temp = this->textId;
+        if ((temp == 0xDBB) || (temp == 0xDBC)) {
+            if (!(gSaveContext.weekEventReg[0x18] & 0x80)) {
```

Comment:
```suggestion
            if (!(gSaveContext.weekEventReg[24] & 0x80)) {
```
We decided that these should be in dec since they're array accesses.